c7n/resources/hsm.py

```diff
@@ -37,19 +37,16 @@ class resource_type(object):
         # Note: resourcegroupstaggingapi still points to hsm-classic
 
     augment = universal_augment
-    _generate_arn = None
 
     @property
     def generate_arn(self):
-        if self._generate_arn is None:
-            self._generate_arn = functools.partial(
-                generate_arn,
-                'cloudhsm',
-                region=self.config.region,
-                account_id=self.account_id,
-                resource_type='cluster',
-                separator='/')
-        return self._generate_arn
+        return functools.partial(
```

Comment:
i'm gonna back this out, i've got something in the works to refactor arn handling across the board.